cpp/src/arrow/compute/kernels/scalar_temporal_test.cc

```diff
@@ -409,6 +409,60 @@ class ScalarTemporalTest : public ::testing::Test {
       RoundTemporalOptions(15, CalendarUnit::QUARTER);
   RoundTemporalOptions round_to_15_years = RoundTemporalOptions(15, CalendarUnit::YEAR);
 
+  RoundTemporalOptions round_to_1_nanoseconds_2 =
+      RoundTemporalOptions(1, CalendarUnit::NANOSECOND, true, true, false);
```

Comment:
Hmm... to avoid the naming issue, perhaps we can have a derived fixture for `ceil_is_strictly_greater = true`?

Reply:
Yes! Done.

Reply:
Thanks... why is `calendar_based_origin` false here? Am I misreading? Or are these unused?

Reply:
Oh that was wrong. Fixed.
The reason the tests passed is because the error (`calendar_based_origin=true`) was only there for `multiple=1` which would return (mostly) equal for `calendar_based_origin=true` or `calendar_based_origin=false`.